### ISSUE TITLE
Preview snackbar too easily dismissed when intending to click preview link

### ISSUE THREAD

**Issue by westonruter** [WordPress/gutenberg#71471]
### Description

When I save a post, I very often want to preview the post. But when I attempt to click "View preview" in the snackbar, I find I often don't click exactly on the link text, and the result is the entire snackbar is dismissed. I have to click exactly on top of the link or the arrow to open the preview:

https://github.com/user-attachments/assets/fb76284f-1307-4a02-97b8-735b94c5cb86

Either clicking anywhere entire snackbar should open the preview, or some padding should be added to the link to make it an easier click target.

Related: 
* https://github.com/WordPress/gutenberg/issues/28479

### Step-by-step reproduction instructions

1. Edit a post.
2. Hit Save Draft
3. Click on or around "View Preview" in the snackbar.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by Mamaduka** [comment]
General note: Snackbar is part of the component system; changes to it can affect apps globally. Related: #34306.
